File: program7/main.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var emailID string
	fmt.Print("Enter email id: ")
	fmt.Scan(&emailID)

	parts := strings.Split(emailID, "@")
	if len(parts) != 2 {
		fmt.Println("No data or @ does not exist")
		return
	}

	p1, p2, p3 := parts[0], "@", parts[1]

	if p1 == "" || p2 == "" || p3 == "" {
		fmt.Println("No data or @ does not exist")
	} else if p1 != "" && p2 == "@" && p3 != "edupillar.com" {
		fmt.Println("Domain does not match")
	} else if p1 == "" && p2 == "@" && p3 == "edupillar.com" {
		fmt.Println("Only Domain")
	} else if p1 != "" && p2 == "@" && p3 == "edupillar.com" {
		fmt.Println("Domain Matches")
	} else {
		fmt.Println("Invalid")
	}
}

File: program7/main.py
email_id=input("Enter email id: ")
p1,p2,p3=email_id.partition("@")
if not (p1) and not(p2) and not(p3):
    print("No data or @ does not exist")
elif p1 and not(p2) and p3:
    print("Invalid")
elif p1 and not(p3=="edupillar.com"):
    print("Domain does not match")
elif not(p1) and (p3=="edupillar.com"):
    print("Only Domain")
elif p1 and (p3=="edupillar.com"):
    print("Domain Matches")
elif not(p1) and p2 and not(p3):
    print("Invalid")
